src/tasks.rs

```diff
@@ -556,4 +556,169 @@ mod tests {
+                            task.update(TaskCode::Normal, &Bytes::from_static(b"updated"));
+                        }
+                    } // Release table lock
+                    tm1.notify_done(&id);
+                });
+
+                // Thread 2: Concurrent notification (could race with first thread)
+                let handle2 = thread::spawn(move || {
+                    tm2.notify_done(&id);
+                });
+
+                handle1.join().unwrap();
+                handle2.join().unwrap();
+            });
+        }
+
+        /// Test concurrent delete and notify operations
+        #[test]
+        fn concurrent_delete_notify() {
+            loom::model(|| {
+                let task_manager = Arc::new(SimpleTaskManager::new());
+                
+                // Pre-populate with a task
+                let id = task_manager.add_task(Bytes::from_static(b"test"));
+                
+                let tm1 = task_manager.clone();
+                let tm2 = task_manager.clone();
+
+                // Thread 1: Delete task
+                let handle1 = thread::spawn(move || {
+                    tm1.delete_task(id);
+                });
+
+                // Thread 2: Notify task done (this could race with delete)
+                let handle2 = thread::spawn(move || {
+                    tm2.notify_done(&id);
+                });
+
+                handle1.join().unwrap();
+                handle2.join().unwrap();
+            });
+        }
+
+        /// Test multiple concurrent operations on different tasks
+        #[test]
+        fn concurrent_multiple_tasks() {
+            loom::model(|| {
+                let task_manager = Arc::new(SimpleTaskManager::new());
+                
+                let tm1 = task_manager.clone();
+                let tm2 = task_manager.clone();
+                let tm3 = task_manager.clone();
+
+                // Thread 1: Add new task
+                let handle1 = thread::spawn(move || {
+                    tm1.add_task(Bytes::from_static(b"task1"));
+                });
+
+                // Thread 2: Add another task  
+                let handle2 = thread::spawn(move || {
+                    tm2.add_task(Bytes::from_static(b"task2"));
+                });
+
+                // Thread 3: Notify a task (may or may not exist)
+                let handle3 = thread::spawn(move || {
+                    tm3.notify_done(&0);
+                });
+
+                handle1.join().unwrap();
+                handle2.join().unwrap();
+                handle3.join().unwrap();
+            });
+        }
+
+        /// Simplified TaskManager for loom testing
+        /// Focuses only on the synchronization patterns that caused deadlocks
+        struct SimpleTaskManager {
+            table: Mutex<HashMap<u32, Task>>,
+            notifiers: Mutex<HashMap<u32, bool>>, // Simplified notifier state
+            current_id: Mutex<u32>,
+        }
+
+        impl SimpleTaskManager {
```

Comment:
DO NOT create a new one. Re-use the old one and follow the instructions in loom doc:

- https://docs.rs/loom/latest/loom/#writing-tests

Also, add loom test to the CI.